### ISSUE TITLE
[RFE] add k3s network option: flannel

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#39128]
**Is your feature request related to a problem? Please describe.**
an option for k3s is flannel. We should allow the user to select from a list of available options (as we do for rke1/2)

**Describe the solution you'd like**
* allow user to select flannel
* support the following flags: https://rancher.com/docs/k3s/latest/en/installation/network-options/#flannel-options

**Describe alternatives you've considered**
n/a

**Additional context**
this will be required for k3s GA

**Comment by snasovich** [comment]
@slickwarren , 

1. I believe "flannel" is what is used by default and in fact is the only supported option
2. `--flannel-backend` flag should already be supported as it's already in serverArgs in KDM - https://github.com/rancher/kontainer-driver-metadata/blob/135eb5b1e118f0003543f5caa0b6c0780477a5f5/channels.yaml#L39; I don't believe `--flannel-ipv6-masq` is added though, but it's probably lower priority
3. `none` option for `cni` should already be supported when editing as YAML; did you try it?

**Comment by slickwarren** [comment]
yes, `none` works fine and is validated through the yaml. We can check those off. I think I'd accidentally created this in rancher instead of dashboard
